### PR TITLE
Show EOL warning in the update section

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref: https://github.com/nextcloud/updater_server/pull/109

> bildschirmfoto von 2018-03-14 14-19-07

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: apps/updatenotification/lib/UpdateChecker.php [hunk 1]
```diff
@@ -49,6 +49,7 @@ public function getUpdateState(): array {
 			$result['updateAvailable'] = true;
 			$result['updateVersion'] = $data['versionstring'];
 			$result['updaterEnabled'] = $data['autoupdater'] === '1';
+			$result['versionIsEol'] = $data['eol'] === '1';
```
Comment (MorrisJobke) [inline apps/updatenotification/lib/UpdateChecker.php]: Does this also work if `eol` is not set? (just asking for old updater servers ;))

**Comment by MorrisJobke** [comment]
Just one question: we only show this warning if the updater server actually returns something. So we need to return a response with EOL for every old release right?

That also means we should ship the EoL flag for an instance on 11.0.8, that gets a response with 12.0.5, right? So we response with EoL based on the source version and not the destination version. Just want to check back on this here.

File: lib/private/Updater/VersionCheck.php [hunk 2]
```diff
@@ -98,6 +98,7 @@ public function check() {
 				$tmp['url'] = (string)$data->url;
 				$tmp['web'] = (string)$data->web;
 				$tmp['autoupdater'] = (string)$data->autoupdater;
+				$tmp['eol'] = isset($data->eol) ? (string)$data->eol : '0';
```
Comment (nickvergessen) [inline lib/private/Updater/VersionCheck.php]: @MorrisJobke isset check for old servers is here

**Comment by nickvergessen** [comment]
> That also means we should ship the EoL flag for an instance on 11.0.8, that gets a response with 12.0.5, right? So we response with EoL based on the source version and not the destination version. Just want to check back on this here.

Yes, that's what I did on the updater server :+1:

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@nickvergessen Failing unit tests

**Comment by rullzer** [comment]
Conflict

**Comment by nickvergessen** [comment]
rebased

**Comment by nickvergessen** [comment]
Should we backport this to 11? @MorrisJobke your call.

**Comment by MorrisJobke** [comment]
> Should we backport this to 11? @MorrisJobke your call.

11 would not make sense, because there will be no future 11 release ;)

But 12 and 13 would be nice

**Comment by nickvergessen** [comment]
Backports in #8986 and #8987

**Pull request merged**